gcm/gcm_connector.go

```diff
@@ -169,18 +163,53 @@ func (gcmConnector *GCMConnector) GetPrefix() string {
 	return gcmConnector.prefix
 }
 
+//func (gcmConnector *GCMConnector) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+//	gcmConnector.mux.ServeHTTP(w, r)
+//}
+
 func (gcmConnector *GCMConnector) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	gcmConnector.mux.ServeHTTP(w, r)
+	if r.Method != http.MethodPost {
+		protocol.Err("Only HTTP POST METHOD SUPPORTED but received type=" + r.Method)
+		http.Error(w, "Permission Denied", 405)
+		return
+	}
+
+	userID, gcmID, topic, err := gcmConnector.parseParams(r.URL.Path)
+	if err != nil {
+		http.Error(w, "Permission Denied", 405)
+		return
+	}
+	gcmConnector.subscribe(topic, userID, gcmID)
+
+	fmt.Fprintf(w, "registered: %v\n", topic)
 }
 
-func (gcmConnector *GCMConnector) Subscribe(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
-	topic := params.ByName(`topic`)
-	userid := params.ByName("userid")
-	gcmid := params.ByName(`gcmid`)
+// parseParams will parse the HTTP URL with format /gcm/:userid/:gcmid/subscribe/*topic
+// returning error if the request is not in the corect format   or else the parsed Params
+func (gcm *GCMConnector) parseParams(path string) (userID, gcmID, topic string, err error) {
+	subscribePrefixPath := "subscribe"
+	currentUrlPath := removeTrailingSlash(path)
 
-	gcmConnector.subscribe(topic, userid, gcmid)
+	if strings.HasPrefix(currentUrlPath, gcm.prefix) != true {
+		return userID, gcmID, topic, errors.New("Gcm request is not starting with gcm prefix")
```

Comment:
Specify the prefix in the error. Also the code can be:
```go
err = errors.New("Gcm request is not starting with gcm prefix")
return
```

Go will return the variables assigned in the function definition